tests/app/main/test_suppliers.py

```diff
@@ -2662,7 +2682,26 @@ def test_fails_if_api_update_fails(self, data_api_client):
             res = self.client.post("/suppliers/registration-number/edit", data=valid_post_data)
             assert res.status_code == 504
 
-    def test_get_shows_already_entered_page_and_api_not_called_update_if_data_already_entered(self, data_api_client):
+    @pytest.mark.parametrize('overwrite_supplier_data',
+                             (
+                                 {'companyDetailsConfirmed': False},
+                                 {'companiesHouseNumber': None, 'otherCompanyRegistrationNumber': None},
+                             ))
+    def test_get_shows_form_on_page_if_supplier_data_not_complete_and_confirmed(self, data_api_client,
```

Comment:
Are test on line 2441 and this test essentially the same test for two different pages? If yes, maybe it would be good to combine them / do a more robust parametrise to test behaviour of all affected pages? What do you think?

Reply:
They are basically the same yeah. I decided not to combine them/parameterize them so that the test code stays close to the rest of the context for the page. There might be more I could do. I'll take another look and see if there's any sensible options to de-duplicate.

Reply:
I see, yeah, it does make sense, but then shouldn't we have one for each page?

Reply:
We do have one of these tests for each of the three pages I think?